result.go

```diff
@@ -22,6 +22,7 @@ type SnowflakeResult interface {
 	GetQueryID() string
 	GetStatus() queryStatus
 	GetArrowBatches() ([]*ArrowBatch, error)
+	GetChunkMetas() ([]ExecResponseChunk, error)
```

Comment:
Instead of adding this to the `SnowflakeResult` interface can it be a new interface? Then we can only implement it for the variant that actually supports it.